sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java

```diff
@@ -235,6 +237,15 @@ public Integer getMaxItemCountForVectorSearch() {
         return this.maxItemCountForVectorSearch;
     }
 
+    /**
+     * Gets the maximum item size to fetch during hybrid search queries.
+     *
+     * @return the max number of items for hybrid search.
+     */
+    public Integer getMaxItemCountForHybridSearch() {
+        return this.maxItemCountForHybridSearch != null ? this.maxItemCountForHybridSearch : Configs.DEFAULT_MAX_ITEM_COUNT_FOR_HYBRID_SEARCH;
```

Comment:
should here be: getMaxItemCountForHybridSearchSearch()?

Reply:
@xinlian12 Why should we have the name of the function with two SearchSearch() in the end?

Reply:
sorry it is typo - I mean Configs.getMaxItemCountForHybridSearch()

Reply:
@aayush3011 - Annie's point is that if not set in QueryRequestOptions it should use the effective default (which could be Configs.DEFAULT_MAX_ITEM_COUNT_FOR_HYBRID_SEARCH if not overridden in system property or env variable - but when overridden it should be honored). So, Annie's proposal to use 

return this.maxItemCountForHybridSearch != null ? this.maxItemCountForHybridSearch : Configs.getMaxItemCountForHybridSearch();

makes sense to me.

Reply:
Made the code changes.